### ISSUE TITLE
bug: dist-custom-elements does not work with ssr

### ISSUE THREAD

**Issue by liamdebeasi** [ionic-team/stencil#3101]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.9.0

### Current Behavior

When building a Stencil project in an SSR environment, the compilation fails with an `HTMLElement is not defined`. This is because the custom elements directly extend `HTMLElement`, but `HTMLElement` is not available in an SSR environment.

So for example, `ion-accordion` component in Ionic 6 looks like this:

```js
const Accordion = class extends HTMLElement { ... }
```

As a workaround, we could have the classes get extended from a special class that works in SSR:

```js
const HTMLElementSSR = (
  typeof HTMLElement !== 'undefined'
  ? HTMLElement
  : class {});

const Accordion = class extends HTMLElementSSR { ... }
```

### Expected Behavior

I would expect the build to not fail.

### Steps to Reproduce

1. Clone https://github.com/liamdebeasi/ionic-next-js. This is a Next.js app running Ionic React 6
2. Run `npm install`.
3. Run `npm run build`. Observe that the build gives an error: `ReferenceError: HTMLElement is not defined`. Opening the source of the error should reference an Ionic component that directly extends `HTMLElement`.

### Code Reproduction URL

https://github.com/liamdebeasi/ionic-next-js

### Additional Information

Related Ionic issue: https://github.com/ionic-team/ionic-framework/issues/24026

**Comment by rwaskiewicz** [comment]
I've confirmed this has been a bug since Stencil@v2.0.0 (at the very least). We'll get this ingested into our issue tracker

**Comment by rwaskiewicz** [comment]
@willmartian's PR, #3131 has been released in [release of v2.11.0](https://github.com/ionic-team/stencil/releases/tag/v2.11.0) of Stencil, which happened ~30 minutes ago. Should anyone find a new instance of it (this bug), please open a new issue!
